app/uk/gov/hmrc/ctcguaranteebalanceeisstub/models/TestScenarios.scala

```diff
@@ -60,17 +60,19 @@ object TestScenarios {
         Right(BalanceResponse(data.grn, Balance(data.amount.doubleValue), data.currency))
     }
 
-  val accessCodeValidationScenarios: Map[GuaranteeReferenceNumber, Map[AccessCode, Either[Result, AccessCodeResponse]]] =
-    simpleGuaranteeTestDataList.map { data =>
-      val grnModel       = data.grn
-      val validCodeModel = AccessCode(data.accessCode)
-
-      val accessCodeMap: Map[AccessCode, Either[Result, AccessCodeResponse]] =
-        Map(validCodeModel -> Right(AccessCodeResponse(grnModel, validCodeModel)))
-          .withDefaultValue(Left(invalidAccessCodeResult))
-
-      grnModel -> accessCodeMap
-    }.toMap
+  def getAccessCodeValidationScenarios(grn: GuaranteeReferenceNumber, providedCode: AccessCode): Either[Result, AccessCodeResponse] = {
```

Comment:
You could make this function return Result rather than an either of Result/AccessCodeResponse this means you can remove logic in the controller that isn't really required